Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/BaseFragment.java

```diff
@@ -133,5 +108,32 @@ public String getDisplayedClassName() {
         return this.getClass().getSimpleName();
     }
 
+    private class TutorialStepTransactionListener implements TransactionListener<TutorialStep> {
```

Comment:
Static innerclasses don't keep a reference to the object, they belong to the class. Making TutorialStepTransactionListener a static inner class will remove the reference however it then requires tutorialText, tutorialTexts and tutorialCanBeDeferred to be static variables as they are needed inside the static innerclass. This means these three variables are then shared among all fragments that derive from BaseFragment and they also become GC roots.The original version (non-static inner class) made the leak canary fault go away. I will change to a static inner and retest with leak canary and check again. Will update with more info when I have it.